File: 백준/Silver/2217. 로프/로프.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.Collections;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        Integer[] rope = new Integer[n];

        for (int i = 0; i < n; i++) {
            rope[i] = Integer.parseInt(br.readLine());
        }

        Arrays.sort(rope, Collections.reverseOrder());

        int weight = 0;
        for (int i = 0; i < n; i++) {
            weight = Math.max(weight, rope[i] * (i + 1));
        }

        System.out.print(weight);
    }
}

File: 백준/Silver/2217. 로프/로프.kt
import java.lang.Integer.max

fun main() {
    val n = readln().toInt()
    val list = ArrayList<Int>(100_002)
    for (i in 0 until n) {
        list.add(readln().toInt())
    }
    list.sort()
    var weight = 0
    for (i in 1..n) {
        weight = max(weight, list[n - i] * i)
    }
    println(weight)
}